### PR TITLE
[tests-only] remove connection

### PR DESCRIPTION
**Pull request by swoichha**
## Description
This PR adds tests for removing connection after a user has been added to the desktop-client

## Related Issue
- https://github.com/owncloud/client/issues/8651

### CODE REVIEW

File: test/gui/shared/scripts/pageObjects/AccountStatus.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        squish.sendEvent(
+            "QMouseEvent",
+            squish.waitForObject(self.ACCOUNT_BUTTON),
+            squish.QEvent.MouseButtonPress,
+            0,
+            0,
+            squish.Qt.LeftButton,
+            0,
+        )
+        squish.activateItem(squish.waitForObjectItem(self.ACCOUNT_MENU, "Log in"))
```
Comment (Talank) [inline test/gui/shared/scripts/pageObjects/AccountStatus.py:56]: I think you can tale the 2nd argument like `action` or something and combine these 2 methods to one. Because these 2 methods do exactly the same thing except for selecting either "login" or "log out".

**Review by Talank — CHANGES_REQUESTED** [review]

File: test/gui/tst_removeAccount/test.feature [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    Given user "Alice" has been created on the server with default attributes and without skeleton files
+    And user "Brian" has been created on the server with default attributes and without skeleton files
+    And user "Alice" has set up a client with default settings
+    And the user has added an account with
+        | server      | %local_server%     |
+        | user        | Brian              |
+        | password    | AaBb2Cc3Dd4        |
+        | localfolder | %client_sync_path% |
+    When the user removes the account
+	Then an account should be displayed with the displayname Alice Hansen and host %local_server_hostname%
```
Comment (Talank) [inline test/gui/tst_removeAccount/test.feature:18]: The spacing seems a little distorted

File: test/gui/shared/scripts/pageObjects/AccountStatus.py (hunk 1 above)
Comment (swoichha) [reply test/gui/shared/scripts/pageObjects/AccountStatus.py]: Done

File: test/gui/tst_removeAccount/test.feature (hunk 2 above)
Comment (swoichha) [reply test/gui/tst_removeAccount/test.feature]: fixed

File: test/gui/shared/scripts/pageObjects/AccountStatus.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            "QMouseEvent",
+            squish.waitForObject(self.ACCOUNT_BUTTON),
+            squish.QEvent.MouseButtonPress,
+            0,
+            0,
+            squish.Qt.LeftButton,
+            0,
+            0,
+        )
+        squish.activateItem(squish.waitForObjectItem(names.settings_QMenu, "Remove"))
```
Comment (Talank) [inline test/gui/shared/scripts/pageObjects/AccountStatus.py:57]: instead of this, you can call loginLogoutAction("Remove"). And it would be nice if you rename the function `loginLogoutAction`, it sounds very specific. May be `accountAction` or something like that. 🤔

Comment (swoichha) [reply test/gui/shared/scripts/pageObjects/AccountStatus.py:57]: Yes, that can be done but we still need to have `removeUser` method as it does more then just clicking on "Remove". I have refactored this method and made it more DRY.

**Review by Talank — COMMENTED** [review]
LGTM

File: test/gui/shared/steps/steps.py [hunk 4]
```diff
@@ -67,6 +67,16 @@ def hook(context):
         pass
 
 
+@Given('the user has added an account with')
+def step(context):
+    clickButton(
+        waitForObject(names.settings_settingsdialog_toolbutton_Add_account_QToolButton)
```
Comment (individual-it) [inline test/gui/shared/steps/steps.py:73]: shouldn't that be part of the page object?

File: test/gui/shared/steps/steps.py [hunk 5]
```diff
@@ -588,3 +598,39 @@ def step(context, username):
...
+    host = substituteInLineCodes(context, host)
+    test.compare(
+        str(
+            waitForObjectExists(
+                {
+                    "name": "settingsdialog_toolbutton_" + displayname + "@" + host,
+                    "type": "QToolButton",
+                    "visible": 1,
+                }
+            ).text
```
Comment (individual-it) [inline test/gui/shared/steps/steps.py:625]: move into a page object if possible

File: test/gui/tst_removeAccount/test.feature [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  Scenario: remove an account
+    Given user "Alice" has been created on the server with default attributes and without skeleton files
+    And user "Brian" has been created on the server with default attributes and without skeleton files
+    And user "Alice" has set up a client with default settings
+    And the user has added an account with
+        | server      | %local_server%     |
+        | user        | Brian              |
+        | password    | AaBb2Cc3Dd4        |
+        | localfolder | %client_sync_path% |
+    When the user removes the account
```
Comment (individual-it) [inline test/gui/tst_removeAccount/test.feature:17]: which account is to be removed? You have set up two

File: test/gui/shared/scripts/pageObjects/AccountStatus.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+            squish.QEvent.MouseButtonPress,
+            0,
+            0,
+            squish.Qt.LeftButton,
+            0,
+            0,
+        )
+        squish.activateItem(squish.waitForObjectItem(self.ACCOUNT_MENU, action))
+
+    def removeUser(self):
```
Comment (individual-it) [inline test/gui/shared/scripts/pageObjects/AccountStatus.py:46]: I'm astonished about how we remove a user through the desktop client ;-)
Maybe call it `removeConnection` or similar ?

**Review by individual-it — CHANGES_REQUESTED** [review]

File: test/gui/tst_removeAccount/test.feature (hunk 6 above)
Comment (swoichha) [reply test/gui/tst_removeAccount/test.feature:17]: This will remove the account that was added later on. In this case the user that will be removed is `Brian`

File: test/gui/shared/steps/steps.py (hunk 4 above)
Comment (swoichha) [reply test/gui/shared/steps/steps.py]: Fixed

File: test/gui/shared/steps/steps.py (hunk 5 above)
Comment (swoichha) [reply test/gui/shared/steps/steps.py]: Done

File: test/gui/shared/scripts/pageObjects/AccountStatus.py (hunk 7 above)
Comment (swoichha) [reply test/gui/shared/scripts/pageObjects/AccountStatus.py]: I have now renamed it to `removeConnection`.

File: test/gui/shared/scripts/names.py [hunk 8]
```diff
@@ -98,3 +96,4 @@
 oCC_ShareLinkWidget_checkBox_expire_QProgressIndicator = {"aboveWidget": oCC_ShareLinkWidget_lineEdit_password_QLineEdit, "container": qt_tabwidget_stackedwidget_OCC_ShareLinkWidget_OCC_ShareLinkWidget, "leftWidget": oCC_ShareLinkWidget_checkBox_expire_QCheckBox, "type": "QProgressIndicator", "unnamed": 1, "visible": 1}
 oCC_IssuesWidget_tableView_QTableView = {"container": qt_tabwidget_stackedwidget_OCC_IssuesWidget_OCC_IssuesWidget, "name": "_tableView", "type": "QTableView", "visible": 1}
 o_tableView_0_1_QModelIndex = {"column": 1, "container": oCC_IssuesWidget_tableView_QTableView, "row": 0, "type": "QModelIndex"}
+owncloudWizard_QWidget = {"type": "QWidget", "unnamed": 1, "visible": 1, "window": owncloudWizard_OCC_OwncloudWizard}
```
Comment (individual-it) [inline test/gui/shared/scripts/names.py:99]: seems to be unused

File: test/gui/shared/scripts/pageObjects/Toolbar.py [hunk 9]
```diff
@@ -9,50 +10,29 @@ class Toolbar:
...
-        squish.sendEvent(
-            "QMouseEvent",
-            squish.waitForObject(self.ACCOUNT_BUTTON),
-            squish.QEvent.MouseButtonPress,
-            0,
-            0,
-            squish.Qt.LeftButton,
-            0,
-            0,
+    def isAccountDisplyed(self, displayname, host):
```
Comment (individual-it) [inline test/gui/shared/scripts/pageObjects/Toolbar.py:26]: better to get the text here and return it, the assertion should happen in the step

also there is a typo in `Displyed`

File: test/gui/shared/steps/steps.py [hunk 10]
```diff
@@ -516,16 +524,16 @@ def step(context, resource, role):
 
 @When('the user logs out of the client-UI')
```
Comment (individual-it) [inline test/gui/shared/steps/steps.py:525]: just seeing now, the "logout" is of-course also per account, so we should define out of which account (if multiple are configured) the users logsout. Can be a separate PR

File: test/gui/shared/steps/steps.py [hunk 11]
```diff
@@ -588,3 +596,30 @@ def step(context, username):
     # It might take some time for all files to sync and connect to ther server
     snooze(5)
     isUserSignedIn(context, username)
+
+
+@When("the user removes the connection")
```
Comment (individual-it) [inline test/gui/shared/steps/steps.py:601]: Which connection? If multiple are configured

maybe add `the last created` or `the currently selected` to the sentence.

**Review by individual-it — CHANGES_REQUESTED** [review]

File: test/gui/tst_removeAccount/test.feature (hunk 6 above)
Comment (individual-it) [reply test/gui/tst_removeAccount/test.feature]: That need to be specified in the step (for less astonishment ;-) )

File: test/gui/shared/scripts/names.py (hunk 8 above)
Comment (swoichha) [reply test/gui/shared/scripts/names.py]: removed

File: test/gui/shared/scripts/pageObjects/Toolbar.py (hunk 9 above)
Comment (swoichha) [reply test/gui/shared/scripts/pageObjects/Toolbar.py]: fixed

File: test/gui/shared/steps/steps.py (hunk 10 above)
Comment (swoichha) [reply test/gui/shared/steps/steps.py:525]: Yes, you're right about this. I will make a new PR for that.

File: test/gui/shared/steps/steps.py (hunk 11 above)
Comment (swoichha) [reply test/gui/shared/steps/steps.py]: I have refactored this step to `@When('the user removes the connection for user "|any|" and host |any|')` so that we can remove connection for any user we want.

**Review by individual-it — APPROVED** [review]
